e2e/cypress/integration/plugins/demo_plugin/webhook_spec.js

```diff
@@ -0,0 +1,232 @@
+ * Download version 0.1.0 from :
+ * https://github.com/mattermost/mattermost-plugin-demo/releases/download/v0.9.0/com.mattermost.demo-plugin-0.1.0.tar.gz
+ * Copy to : ./e2e/cypress/fixtures/com.mattermost.demo-plugin-0.9.0.tar.gz
+ */
+
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
+describe('Demo plugin - Webhook events', () => {
+    const pluginId = 'com.mattermost.demo-plugin';
+    const pluginName = 'com.mattermost.demo-plugin-0.9.0.tar.gz';
+
+    let testTeam;
+    let testUser;
+    let admin;
+    let testChannel;
+    let newTeam;
+
+    before(() => {
+        // # Set plugin settings
+        const newSettings = {
+            PluginSettings: {
+                Enable: true,
+                RequirePluginSignature: false,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        cy.apiInitSetup().then(({team, user}) => {
+            admin = user;
+            testTeam = team;
+
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+
+            // # Create team before installing demo plugin
+            cy.apiCreateTeam('team', 'Team').then(({team: anotherTeam}) => {
+                newTeam = anotherTeam;
+            });
+
+            // # Uninstall all plugins
+            cy.apiGetAllPlugins().then(({plugins}) => {
+                const {active, inactive} = plugins;
+                inactive.forEach((plugin) => cy.apiRemovePluginById(plugin.id));
+                active.forEach((plugin) => cy.apiRemovePluginById(plugin.id));
+            });
+
+            // # Install demo plugin
+            cy.apiUploadPlugin(pluginName).then(() => {
+                cy.apiEnablePluginById(pluginId);
+            });
+
+            // # Enable webhook events
+            cy.postMessage('/demo_plugin true');
+            cy.wait(TIMEOUTS.TWO_SEC);
+
+            // * Verify that hooks are enabled
+            cy.getLastPost().contains('enabled', {matchCase: false});
+
+            // # Create channel
+            cy.apiCreateChannel(team.id, 'my_test_channel', 'my_test_channel').then(({channel}) => {
+                testChannel = channel.name;
+            });
+
+            // # Join the demo channel
+            cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+
+            // # Create another user
+            cy.apiCreateUser().then(({user: otherUser}) => {
+                testUser = otherUser;
+                cy.apiAddUserToTeam(team.id, testUser.id);
+
+                // # Login another user
+                cy.apiLogin(testUser);
+
+                // # Join the demo channel
+                cy.visit(`/${testTeam.name}/channels/demo_plugin`);
+            });
+        });
+    });
+
+    it('MM-T2408/2 - User posts a message Webhook event', () => {
+        // # Post message
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.postMessage('some message');
```

Comment:
Nit pick: Should be use `import * as MESSAGES from '../../../fixtures/messages';` here as the content is not important?